Mathlib/GroupTheory/Index.lean

```diff
@@ -504,26 +509,43 @@ lemma _root_.AddSubgroup.relindex_toSubgroup {G : Type*} [AddGroup G] (H K : Add
 
 section FiniteIndex
 
-variable (H K)
-
 /-- Typeclass for finite index subgroups. -/
-class FiniteIndex : Prop where
-  /-- The subgroup has finite index -/
+class _root_.AddSubgroup.FiniteIndex {G : Type*} [AddGroup G] (H : AddSubgroup G) : Prop where
+  /-- The additive subgroup has finite index -/
   finiteIndex : H.index ≠ 0
 
+variable (H) in
 /-- Typeclass for finite index subgroups. -/
-class _root_.AddSubgroup.FiniteIndex {G : Type*} [AddGroup G] (H : AddSubgroup G) : Prop where
-  /-- The additive subgroup has finite index -/
+@[to_additive] class FiniteIndex : Prop where
+  /-- The subgroup has finite index -/
   finiteIndex : H.index ≠ 0
 
+/-- Typeclass for a subgroup `H` to have finite index in a subgroup `K`. -/
+class _root_.AddSubgroup.IsFiniteRelIndex {G : Type*} [AddGroup G] (H K : AddSubgroup G) :
+    Prop where
+  protected relIndex_ne_zero : H.relindex K ≠ 0
```

Comment:
Can you explain why having this, and having it as a typeclass is useful? Perhaps @kbuzzard has thoughts too.

Reply:
This is useful to state the following:

https://github.com/ImperialCollegeLondon/FLT/blob/c51f2573c22ded973f354f56f598507b0f349f51/FLT/Mathlib/MeasureTheory/Group/Action.lean#L57-L59

namely that the Haar measure of a subgroup `H` of finite index inside a subgroup `K` is the Haar measure of `K` divided by the index.

A possible alternative would be to have some `ENat`-valued index

Reply:
Please also add this to the PR description.